### PR TITLE
IndexSeries tutorial

### LINKED ISSUE #1601: [Documentation]: Add tutorial on using IndexSeries with Images
### What would you like changed or added to the documentation and why?

Images and IndexSeries were recently enhanced to work together, but there currently isn't a good tutorial for how to do this. We should add a section to the existing Images tutorial in PyNWB

### Do you have any interest in helping write or edit the documentation?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this change was not already [requested](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

### PR DESCRIPTION
**Pull request by bendichter**
## Motivation

fix #1601 

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

File: docs/gallery/domain/images.py [hunk 1]
```diff
@@ -245,6 +245,58 @@
 nwbfile.add_acquisition(images)
 
 ####################
+# IndexSeries for repeated images
+# -------------------------------
+#
+# You may want to set up a time series of images where some images are repreated many
```
Comment (CodyCBakerPhD) [inline docs/gallery/domain/images.py]:
```suggestion
# You may want to set up a time series of images where some images are repeated many
```

File: docs/gallery/domain/images.py [hunk 2]
```diff
@@ -245,6 +245,58 @@
...
+# IndexSeries for repeated images
+# -------------------------------
+#
+# You may want to set up a time series of images where some images are repeated many
+# times. You could create an :py:class:`~pynwb.image.ImageSeries` that repeats the data
+# each time the image is shown, but that would be inefficient, becuase it would store
+# the same data multiple times. A better solution would be to store the unique images
+# once and reference those images. This is how :py:class:`~pynwb.image.IndexSeries`
+# works. First, create an :py:class:`~pynwb.base.Images` container with the order of
+# images defined using a :py:class:`~pynwb.base.ImageReferences`. Then create an
```
Comment (CodyCBakerPhD) [inline docs/gallery/domain/images.py]:
```suggestion
# images defined using an :py:class:`~pynwb.base.ImageReferences`. Then create an
```

File: docs/gallery/domain/images.py [hunk 3]
```diff
@@ -245,6 +245,58 @@
...
+    images=[rgb_face, gs_face],
+    description="A collection of raccoons.",
+    order_of_images=ImageReferences("order_of_images", [rgb_face, gs_face]),
+)
+
+idx_series = IndexSeries(
+    name="stimuli",
+    data=[0, 1, 0, 1],
+    indexed_images=images,
+    unit="N/A",
```
Comment (CodyCBakerPhD) [inline docs/gallery/domain/images.py:292]: Couple things here;

1) should this be `n.a.`? I think we've usually followed that standard (also that's how other units in this gallery are specified)

2) For something called an `IndexSeries`, it seems so weird to even have units at all, and feels weirder to even be required to specify it. What would you think about removing it or at least relaxing it to be optional (and default to None or something)?

File: src/pynwb/image.py [hunk 4]
```diff
@@ -212,23 +212,49 @@ class IndexSeries(TimeSeries):
     array indicates when that image was displayed.
     '''
 
-    __nwbfields__ = ('indexed_timeseries',)
+    __nwbfields__ = (
+        "indexed_timeseries",
+        "indexed_images"
+    )
```
Comment (CodyCBakerPhD) [inline src/pynwb/image.py]: Was this missing `nwbfield` a bug? Should tests be added for it?

File: docs/gallery/domain/images.py (hunk 3 above)
Comment (bendichter) [reply docs/gallery/domain/images.py:292]: Yeah, actually you need to specify it. It's a backwards compatibility quark: https://github.com/NeurodataWithoutBorders/pynwb/blob/e03b0a932c4f83c9f143d4a96973e52219ce7917/src/pynwb/image.py#L234-L236

File: src/pynwb/image.py (hunk 4 above)
Comment (bendichter) [reply src/pynwb/image.py]: I'm not sure. Actually let me try to remove this and see if everything still works

File: docs/gallery/domain/images.py (hunk 3 above)
Comment (CodyCBakerPhD) [reply docs/gallery/domain/images.py:292]: Hmm wow OK lol

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**